File: tomcat/src/spg/models/Country.java
package spg.models;

import org.json.JSONObject;

/**
 * Model for a Country.
 * 
 * @author Paul Barnhill 
 * @version 2018-04-18
 */
public class Country {
	private final String countryID;
	private final String country;
	
	public Country(String countryCode, String country) {
		this.countryID = countryCode;
		this.country = country;
	}
	
	public String getCountryCode() {
		return this.countryID;
	}
	
	public String getCountryName() {
		return this.country;
	}
	
	/**
	 * Returns a JSONObject representation of this Country
	 */
	public JSONObject toJSON() {
		JSONObject j = new JSONObject();
		j.put("countryID", this.countryID);
		j.put("country", this.country);
		return j;
	}
	
	/**
	 * Returns a JSONObject String representation of this Country.
	 */
	public String toString() {
		return this.toJSON().toString();
	}
}

File: node/src/models/country.ts
/**
 * @author Paul Barnhill
 * @version 2018-04-16
 */
import SpgModel from '@src/models/SpgModel';

/**
 * Properties of a country.
 */
export interface Properties {
	countryID: string;
	country: string;
}

/**
 * Data model for a country.
 */
export class Country extends SpgModel {
	public readonly countryID: string;
	public readonly country: string;

	constructor(props: Properties) {
		super();
		if (!props.countryID.length) {
			throw Error('countryID cannot be empty.');
		}
		if (!props.country.length) {
			throw Error('country cannot be empty.');
		}

		this.countryID = props.countryID;
		this.country = props.country;
	}

	toProperties(): Properties {
		return {
			countryID: this.countryID,
			country: this.country
		};
	}
}
